tests/MongoDB.Driver.Tests/AggregateFluentTests.cs

```diff
@@ -1032,46 +1090,58 @@ public void ReplaceWith_should_add_the_expected_stage([Values(false, true)] bool
+        private IAggregateFluent<C> CreateCollectionSubject(IClientSessionHandle session = null, IMongoCollection<C> collection = null)
         {
-            CreateMockCollection();
+            collection = collection ?? CreateMockCollection().Object;
 
             var emptyPipeline = new EmptyPipelineDefinition<C>();
             var options = new AggregateOptions();
-            return new CollectionAggregateFluent<C, C>(session, _mockCollection.Object, emptyPipeline, options);
+            return new CollectionAggregateFluent<C, C>(session, collection, emptyPipeline, options);
         }
 
-        private IAggregateFluent<NoPipelineInput> CreateDatabaseSubject(IClientSessionHandle session = null)
+        private IAggregateFluent<NoPipelineInput> CreateDatabaseSubject(IClientSessionHandle session = null, IMongoDatabase database = null)
         {
-            CreateMockDatabase();
+            database = database ?? CreateMockDatabase().Object;
 
             var emptyPipeline = new EmptyPipelineDefinition<NoPipelineInput>(NoPipelineInputSerializer.Instance);
             var options = new AggregateOptions();
-            return new DatabaseAggregateFluent<NoPipelineInput>(session, _mockDatabase.Object, emptyPipeline, options);
+            return new DatabaseAggregateFluent<NoPipelineInput>(session, database, emptyPipeline, options);
         }
 
-        private void CreateMockCollection()
+        private Mock<IMongoCollection<C>> CreateMockCollection(IMongoDatabase database = null, string collectionName = "test")
         {
-            CreateMockDatabase();
+            database = database ?? CreateMockDatabase().Object;
+
+            var databaseNamespace = database.DatabaseNamespace;
+            var collectionNamespace = new CollectionNamespace(databaseNamespace, collectionName);
 
             var settings = new MongoCollectionSettings();
-            _mockCollection = new Mock<IMongoCollection<C>>();
-            _mockCollection.SetupGet(c => c.Database).Returns(_mockDatabase.Object);
-            _mockCollection.SetupGet(c => c.DocumentSerializer).Returns(settings.SerializerRegistry.GetSerializer<C>());
-            _mockCollection.SetupGet(c => c.Settings).Returns(settings);
+            var mockCollection = new Mock<IMongoCollection<C>>();
+            mockCollection.SetupGet(c => c.CollectionNamespace).Returns(collectionNamespace);
+            mockCollection.SetupGet(c => c.Database).Returns(database);
+            mockCollection.SetupGet(c => c.DocumentSerializer).Returns(settings.SerializerRegistry.GetSerializer<C>());
+            mockCollection.SetupGet(c => c.Settings).Returns(settings);
+
+            return mockCollection;
         }
 
-        private void CreateMockDatabase()
+        private Mock<IMongoDatabase> CreateMockDatabase(string databaseName = "test")
         {
-            _mockDatabase = new Mock<IMongoDatabase>();
```

Comment:
We can no longer assume that only one mock database is sufficient, therefore instead of storing the mock database in a field return the mock database and let the caller decide what to do with it.